src/components/Sweep/SweepApprove.tsx

```diff
@@ -0,0 +1,143 @@
+import { useCallback, useState, useEffect } from 'react';
+import { Form, Input, Button, notification, Typography } from 'antd';
+import { ethers } from 'ethers';
+import styled from 'styled-components';
+
+import { numberToBn, ethereumAddressValidator } from '../../utils';
+import { useMetamask } from '../../hooks/useMetamask';
+import { ChainId } from '../../constants';
+import { contractApprove, contractBalanceOf } from './useSweep';
+
+function SweepApprove() {
+  const [loading, setLoading] = useState(false);
+  const [valid, setValid] = useState(false);
+
+  const [spenderAddr, setSpenderAddr] = useState('');
+  const [approveAmount, setApproveAmount] = useState(0);
+  const [maxAmount, setMaxAmount] = useState(0);
+
+  const [form] = Form.useForm();
+
+  const { changeNetwork, connect } = useMetamask();
+
+  const handleSubmit = useCallback(async () => {
+    try {
+      setLoading(true);
+
+      await form.validateFields();
+
+      await changeNetwork(ChainId.arbitrum);
+      const { account } = await connect();
+
+      if (approveAmount.toString() === '') {
+        await contractApprove(spenderAddr, account!, ethers.constants.MaxUint256);
+      } else {
+        await contractApprove(spenderAddr, account!, numberToBn(approveAmount));
+      }
+
+      notification.success({
+        message: `Wallet ${spenderAddr} has been whitelisted for ${approveAmount} amount successfully. `,
+      });
+    } catch (e) {
+      console.error(e);
+      const error = (e as any)?.message || 'Error';
+      notification.error({ message: error });
+    } finally {
+      setLoading(false);
+    }
+  }, [spenderAddr, approveAmount, changeNetwork, form, connect]);
+
+  const handleSpenderAddrChange = useCallback(
+    (e) => {
+      setSpenderAddr(e.target.value);
+    },
+    [setSpenderAddr]
+  );
+
+  const handleAmountChange = useCallback(
+    (e) => {
+      setApproveAmount(e.target.value);
+    },
+    [setApproveAmount]
+  );
+
+  const setAmountToMax = () => setApproveAmount(maxAmount);
+
+  useEffect(() => {
```

Comment:
have you checked how often this gets called? I know in the case of addresses people just copy paste them 99.9999% of the time but it's good practice to think about those things and handle them appropriately. so imagine that someone was typing into this field - this function would get triggered on every change, which could be multiple times a second. so what you should do is `debounce` this call (500ms should be just about enough). you can example usage e.g. here https://github.com/chronologic/automate-ui/blob/main/src/components/Transactions/tableColumns.tsx#L5 but be aware that you need some extra code for debounce to work properly in `useEffect` https://stackoverflow.com/a/61786423